cpp/include/Ice/StringConverter.h

```diff
@@ -59,7 +59,7 @@ namespace Ice
         virtual ~BasicStringConverter() {}
     };
 
-#if defined(__clang__) && !defined(ICE_STATIC_LIBS)
```

Comment:
Looking at ICE-7473, the `!defined(ICE_STATIC_LIBS)` is not because this code doesn't work with static libs, but because it's not necessary with static libraries.